mlir/lib/Dialect/Vector/IR/VectorOps.cpp

```diff
@@ -6308,6 +6308,48 @@ bool WarpExecuteOnLane0Op::areTypesCompatible(Type lhs, Type rhs) {
       verifyDistributedType(lhs, rhs, getWarpSize(), getOperation()));
 }
 
+//===----------------------------------------------------------------------===//
+// InterleaveOp
+//===----------------------------------------------------------------------===//
+
+// The rank 1 case of vector.interleave on fixed-size vectors is equivalent to a
+// vector.shuffle, which (as an older op) is more likely to be matched by
+// existing pipelines.
+struct FoldRank1FixedSizeInterleaveOp : public OpRewritePattern<InterleaveOp> {
```

Comment:
I think there is a point in using the interleave op as the canonical way to represent this shuffle pattern. It provides more information and it will keep it consistent with scalable.

Reply:
It should be ok to change an existing canonical representation if the new one is more explicit for a particular case.

> I don't think vector.interleave provides any more information in the case either, once you've matched the shuffle mask

Well, that's exactly the point! Some of the ops we have at this level are intended to simplify the analysis part of well-defined cases that are commonly used. Analyzing the shuffle mask is doable but it's adding complexity and we would have to exclude the non-interleave cases from the interleave ones when we have rewrites only targeting the interleave ones. 

> I would not be surprised if adding a vector.shuffle -> vector.interleave canonicalization broke something somewhere (maybe in a downstream project).

It's possible but `vector.shuffle` is used very late in the pipeline and we don't have so many rewrites targeting it. I think the benefit of having scalable and non-scalable aligned and more explicit representation for vector interleave patterns is worth the risk.

Reply:
Sure, that makes sense, I'll try adding the inverse fold and see what breaks :)